### PR TITLE
feature: tabs

### LINKED ISSUE #569: Tabs
It would be nice to have tabs where the content and navigation item are together so you can disable a tab or specify if a user is allowed to see a tab in one place without having to put the code all over the place. The content should also be suspended to improve performance.

### PR DESCRIPTION
**Pull request by gidomanders**
It would be nice to have tabs where the content and navigation item are
together so you can disable a tab or specify if a user is allowed to see
a tab in one place without having to put the code all over the place.
The content should also be suspended to improve performance.

Added Tabs component.

Closes #569

### CODE REVIEW

File: src/core/Tabs/Tabs.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+import { Tabs } from './Tabs';
+
+import { Form } from 'react-final-form';
+import { Tab, Props as TabProps } from './Tab/Tab';
+import Button from '../Button/Button';
+import { JarbInput } from '../../form/Input/Input';
+import { JarbTextarea } from '../../form/Textarea/Textarea';
+
+const tabs: TabProps[] = [
```
Comment (MrHus) [inline src/core/Tabs/Tabs.stories.tsx:12]: Inline this into each story, this makes the stories difficult to copy paste, also this stimulates our users using arrays which is not needed.

File: src/core/Tabs/Tabs.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    active: false,
+    // eslint-disable-next-line react/display-name
+    content: () => <p>This is suspended content</p>,
+    disabled: true,
+    show: () => true,
+    icon: 'warning',
+    iconColor: 'primary'
+  },
+  {
+    label: 'testing',
```
Comment (MrHus) [inline src/core/Tabs/Tabs.stories.tsx:39]: Give this tab a different name

File: src/core/Tabs/Tab/Tab.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  show = alwaysTrue,
+  disabled,
+  iconColor
+}: Props) {
+  if (!show()) {
+    return null;
+  }
+
+  return (
+    <NavItem>
```
Comment (MrHus) [inline src/core/Tabs/Tab/Tab.tsx:79]: For some reason there is no cursor pointer when hovering over the tabs.

File: src/core/Tabs/Tabs.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+storiesOf('core/Tabs', module)
+  .addParameters({ component: Tabs })
+  .addParameters({ component: Tab })
+  .add('default', () => {
+    const [activeTab, setActiveTab] = useState(0);
+    return (
+      <Tabs>
+        <Tab
+          onClick={() => setActiveTab(0)}
+          active={activeTab === 0}
```
Comment (MrHus) [inline src/core/Tabs/Tabs.stories.tsx:61]: Make this a predefined type which is a string. This way people can easier see how to use it in combination with a url param. 

`type Tabs = 'info' | 'testing' | 'button'`.

File: src/core/Tabs/Tab/Tab.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+   *
+   * We highly recommend adding the tab as a URL parameter,
+   * especially if the inactive tabs are not hidden by CSS.
+   */
+  onClick: () => void;
+
+  /**
+   * Callback to render the tab content.
+   */
+  content: () => React.ReactNode;
```
Comment (MrHus) [inline src/core/Tabs/Tab/Tab.tsx:24]: Make this the `children` instead, make sure you use it like children in the stories as well.

```tsx
<Tab>
  {() => <p>Content here></p>}
</Tab>
```

File: src/core/Tabs/Tabs.scss [hunk 6]
```diff
@@ -0,0 +1,13 @@
+.tabs {
+  .nav-item.disabled {
+    cursor: not-allowed;
+  }
+
+  .nav-link {
+    cursor: pointer;
+
+    &.active {
+      cursor: initial;
```
Comment (MrHus) [inline src/core/Tabs/Tabs.scss:10]: Just use cursor pointer for the active one as well, even though nothing will happen.

File: src/core/Tabs/Tabs.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+import { JarbTextarea } from '../../form/Textarea/Textarea';
+
+storiesOf('core/Tabs', module)
+  .addParameters({ component: Tabs })
+  .addParameters({ component: Tab })
+  .add('default', () => {
+    type AvailableTabs = 'info' | 'testing' | 'details';
+    const [activeTab, setActiveTab] = useState<AvailableTabs>('info');
+    return (
+      <Tabs>
```
Comment (MrHus) [inline src/core/Tabs/Tabs.stories.tsx:19]: Please wrap the stories in a `Card` which will make them a little better looking.

File: src/core/Tabs/Tabs.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  });
+
+  return (
+    <div className={classNames('tabs', className)}>
+      <Nav tabs fill style={{ overflowY: 'hidden' }} className="flex-nowrap">
+        {children}
+      </Nav>
+      <div
+        className={classNames('mt-3', {
+          'tab-content': hideInactiveTabsBy === 'CSS'
```
Comment (MrHus) [inline src/core/Tabs/Tabs.tsx:75]: I wonder why we do this? I think adding `tab-content` CSS class will just always work, regardless of the `hideInactiveTabsBy` state.

File: src/core/Tabs/TabContent/TabContent.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  if (hideInactiveTabsBy === 'excluding-from-dom' && !active) {
+    return null;
+  }
+
+  return (
+    <div
+      className={classNames({
+        'tab-pane': hideInactiveTabsBy === 'CSS',
```
Comment (MrHus) [inline src/core/Tabs/TabContent/TabContent.tsx:30]: I wonder why we do this? I think adding `tab-pane` CSS class will just always work, regardless of the `hideInactiveTabsBy` state.

File: src/core/Tabs/Tabs.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+                )}
+              </Tab>
+            </Tabs>
+          </>
+        )}
+      />
+    );
+  });
+
+function sleep(ms: number) {
```
Comment (MrHus) [inline src/core/Tabs/Tabs.stories.tsx:217]: We should put this in `story-utils.tsx`

File: src/core/Tabs/Tabs.tsx (hunk 8 above)
Comment (gidomanders) [reply src/core/Tabs/Tabs.tsx:75]: With tab-content, the tab-pane inside will be hidden unless it has the active class. That doesn't really matter with exclude-from-dom, as long as we always use the active class, so I'll change that.

**Review by MrHus — APPROVED** [review]

**Pull request merged**